x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateFormat.java

```diff
@@ -55,7 +55,7 @@ public class DateFormat extends EsqlConfigurationFunction implements OptionalArg
     )
     public DateFormat(
         Source source,
-        @Param(optional = true, name = "dateFormat", type = { "keyword", "text" }, description = """
```

Comment:
The first positional argument can be a date, in the one parameter version of the function.

Reply:
It's quite a weird way to do it, but that's how the tests work.

We could, optionally, hack around this in the tests *somehow*. Or teach them about first position optional arguments. It's just that we don't have that many!